pkg/executor/core/executor.go

```diff
@@ -129,3 +138,15 @@ func (ex Executor) verifyReference(ctx context.Context, subjectRef common.Refere
 
 	return types.VerifyResult{IsSuccess: isSuccess, VerifierReports: verifyResults}
 }
+
+func (ex Executor) resolveTagToDigest(ctx context.Context, subjectRef common.Reference) (digest.Digest, error) {
+	for _, referrerStore := range ex.ReferrerStores {
+		dig, err := referrerStore.ResolveTag(ctx, subjectRef)
+		if err == nil {
+			return dig, nil
+		}
+		//TODO Add logging for failed resolve calls
```

Comment:
I think this needs to be resolved before merging. Would it be harder to diagnose a 404 without this?

Reply:
Added the log